src/storage/store/column_chunk_data.cpp

```diff
@@ -388,14 +390,17 @@ void ColumnChunkData::setToInMemory() {
     }
 }
 
-void ColumnChunkData::resize(uint64_t newCapacity) {
+void ColumnChunkData::resize(uint64_t newCapacity, bool reserveDataAndInitializeToZero) {
```

Comment:
I don't think this name very clearly indicates what's occurring here.
Maybe this should be two different functions (particularly since, aside from adding the argument name in a comment, boolean arguments are not generally very easy to read). 

While admittedly we haven't been doing this, maybe we should follow std containers like `std::vector` and have a `reserve` function which resizes the buffer without changing the numElements and copies the original data into the buffer (i.e. `reserveDataAndInitializeToZero=true`), and then a separate function that just reallocates a larger buffer without preserving the original data (`reserveDataAndInitializeToZero=false`).

Reply:
Good advice!